Orm/Xtensive.Orm/Sql/Compiler/SqlCompiler.cs

```diff
@@ -37,13 +38,13 @@ public class SqlCompiler : SqlDriverBound,
     /// <param name="unit">Instance to compile.</param>
     /// <param name="compilerConfiguration">Configuration for compiler.</param>
     /// <returns></returns>
-    public SqlCompilationResult Compile(ISqlCompileUnit unit, SqlCompilerConfiguration compilerConfiguration)
+    public SqlCompilationResult Compile(ISqlCompileUnit unit, SqlCompilerConfiguration compilerConfiguration, TypeIdRegistry typeIdRegistry = null)
```

Comment:
This is also not ok, you have a configuration that wraps additional info for compilation. But no upper-level types should be used there.